### PR TITLE
Block bundling: introduce single variation constant

### LINKED ISSUE #28227: Beta blocks: Not available when JETPACK_EXPERIMENTAL_BLOCKS is defined
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Start with JN site
2. In `/wp-admin/options-general.php?page=companion_settings` enable `JETPACK_BETA_BLOCKS`
3. Go to a block editor and search for `beta` blocks. There should be few in the list
4. Go back to `/wp-admin/options-general.php?page=companion_settings` and enable `JETPACK_EXPERIMENTAL_BLOCKS`
5. In the block editor, search for beta blocks

### A clear and concise description of what you expected to happen.

I was expecting that both constants could coexist together

### What actually happened

`JETPACK_EXPERIMENTAL_BLOCKS` overrides block variation that was defined by `JETPACK_BETA_BLOCKS`

### Browser

_No response_

### Other information

The logic is defined here: https://github.com/Automattic/jetpack/blob/5463f2e0284ca8a634f75deee22db0c1bf1de984/projects/plugins/jetpack/class.jetpack-gutenberg.php#L866

It seems that this behavior is expected, but I feel it confusing, especially from the testing perspective.

### Platform (Simple, Atomic, or both?)

_No response_

### Reproducibility

Consistent

### Severity

None

### Available workarounds?

None

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #28227

## Proposed changes:

Instead of having 2 separate constants to manage what block bundle is loaded on a site, let's only use one, `JETPACK_BLOCKS_VARIATION`. It makes it easier to manage and overwrite.

I also sunset 2 existing filters in favor of the existing `jetpack_blocks_variation` filter which works the same way as our new constant.

> **Note**
> Once this gets merged, I'll p2 the change so that everyone contributing to Jetpack can start using the new constant in their dev environments. I've also opened https://github.com/Automattic/companion/pull/61 and 1229-gh-wpcomsh to bring the new constant to our existing environment.

> **Note**
> Since the constants are used by many people, I opted not to display a deprecation notice for those constants just yet. I think it would be noisy. Happy to hear other thoughts on that though.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

There are different things one can test here:

* Start with a brand new JN site.
* Go to Settings > Jetpack constants.
* Toggle the experimental or beta checkboxes.
* Go to Posts > Add New and check the network tab; you should see a matching `editor-beta.js` or `editor-experimental.js` being loaded.
* Now go back to the Jetpack Constants settings page, and disable the checkbox you checked.
* In a functionality plugin or in your theme's `functions.php` file, add the following: `add_filter( 'jetpack_load_beta_blocks', '__return_true' );`
* Go back to Posts > Add New; the Beta bundle should be loaded.
* Now check your logs at `wp-content/debug.log`; you should see a deprecation warning.
* Remove that filter you added earlier.
* Edit your site's `wp-config.php` and add the following: `define( 'JETPACK_EXPERIMENTAL_BLOCKS', 'true' );`
* Back in the post editor, you should see the experimental bundle
* Edit your site's `wp-config.php` again, and add the following instead: `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );`
* Back in the post editor, you should see the beta bundle loaded.
* Go to Settings > Writing
* You should see the checkbox to enable writing prompts towards the top of the page.
* Now install the Jetpack VideoPress plugin
* In the post editor, you should see the Beta VideoPress block.

### CODE REVIEW

File: projects/plugins/jetpack/class.jetpack-gutenberg.php
Comment (jeherve) [inline projects/plugins/jetpack/class.jetpack-gutenberg.php:858]: This is all a bit more complex than I am really happy, but I kept in mind that this should become simpler in a few months once we can remove the deprecated things and the constants. The way this is currently implemented, I think it should be simple for anyone to remove what's not needed anymore.

File: projects/packages/videopress/changelog/update-better-block-variations
Comment (danielbachhuber) [inline projects/packages/videopress/changelog/update-better-block-variations]:
```suggestion
Block bundling: sunset existing methods in favor of new `JETPACK_BLOCKS_VARIATION` constant
```

File: projects/plugins/jetpack/changelog/update-better-block-variations
Comment (danielbachhuber) [inline projects/plugins/jetpack/changelog/update-better-block-variations]:
```suggestion
Block bundling: sunset existing methods in favor of new `JETPACK_BLOCKS_VARIATION` constant
```

**Review by danielbachhuber — COMMENTED** [review]
👍 Makes sense to me conceptually. I'll defer to others on whether these are all of the locations to update.

**Review by coder-karen — DISMISSED** [review]
This works for me, however blocks/blogging-prompts/settings.php has been deleted in trunk (deleted here - https://github.com/Automattic/jetpack/pull/28387/files#diff-763698320ed9f88190022d59f425c61918399a389f5ce93aece51a4172506dba ) so fixing the merge conflict would bring it back again. I presume if it's deleted now that there is no need to make the edits here now.

**Comment by jeherve** [comment]
> I presume if it's deleted now that there is no need to make the edits here now.

Yup, indeed. I've updated the branch with latest trunk now. It should be ready for another review!

@retrofox We talked about it on Slack, did you have a chance to look at this PR? I wouldn't break y'all's workflow or something. :)

**Review by coder-karen — DISMISSED** [review]
Approving, but don't know if you want to wait on @retrofox for a response so left the labels as they are.

**Review by kraftbj — APPROVED** [review]
Approving based on Karen's earlier review. Change was only ver number related.

**Pull request merged**
